selectrum.el

```diff
@@ -1515,16 +1515,22 @@ specs."
   ;; <https://github.com/raxod502/selectrum/issues/21>
   ;; <https://github.com/raxod502/selectrum/issues/58>
   ;; <https://github.com/raxod502/selectrum/pull/76>
-  (setq str (copy-sequence str))
-  (if (version< emacs-version "27")
-      (font-lock-prepend-text-property
+  (let ((str (copy-sequence str))
+        (face 'selectrum-current-candidate))
+    (if (version< emacs-version "27")
+        (font-lock-prepend-text-property
+         0 (length str)
+         'face face str)
+      (add-face-text-property
        0 (length str)
-       'face face str)
-    (add-face-text-property
-     0 (length str)
-     face
-     'append str))
-  str)
+       face
+       'append str)
+      ;; Prepend the background to ensure selection is always visible.
+      (add-face-text-property
+       0 (length str)
+       `(:background ,(face-attribute face :background nil 'inherit))
```

Comment:
Did you mean to use `t` instead? Using `'inherit` will complain about non-existing `inherit` face.

```diff
-       `(:background ,(face-attribute face :background nil 'inherit))
+       `(:background ,(face-attribute face :background nil t))
```
It doesn't fix my issue unfortunately. This still shows red and green background for the current line.

```lisp
(custom-theme-set-faces 'zenburn
  `(selectrum-current-candidate ((t (:inherit 'hl-line)))))

(completing-read "Check: " (list (propertize "this" 'face '(:background "red"))
                                 (propertize "out" 'face '(:background "green"))))
```

If I instead set the background specifically, then I correctly see `zenburn-fg-1` background for the currently selected candidate.

```diff
-  `(selectrum-current-candidate ((t (:inherit 'hl-line)))))
+  `(selectrum-current-candidate ((t (:background ,zenburn-fg-1)))))
```

But according to the documentation for `face-attribute`, using `t` *should* merge the background property (in what priority order that I don't know) so I'm not sure what is wrong here.

And lastly, if I specify the face directly then I get the wanted behavior without having to specify the background color for the Selectrum face specifically:

```diff
-       `(:background ,(face-attribute face :background nil 'inherit))
+       `(:background ,(face-attribute face :background nil 'hl-line))
```

```lisp
(custom-theme-set-faces 'zenburn
  `(selectrum-current-candidate ((t (:inherit 'hl-line)))))
```

Reply:
My bad I should have read the docs of the INHERIT argument, I assumed a non-nil value just meant to get any inherited value, too.

Reply:
> Using 'inherit will complain about non-existing inherit face.

It doesn't for me, this sounds bad does this break Selectrum usage for you?

Reply:
To quickly fix the undefined face problem you get I merged #433.

Reply:
> It doesn't for me, this sounds bad does this break Selectrum usage for you? Which Emacs version are you using?

Yup all selectrum completion can't yield any result because of it:

```
Error in post-command-hook (selectrum--minibuffer-post-command-hook): (error "Invalid face" inherit)
```

Candidates can still be accessed through <kbd>C-p</kbd> and <kbd>C-n</kbd> though.

Emacs version:
```
GNU Emacs 28.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.24, cairo version 1.17.4) of 2021-01-19
```

> I assumed a non-nil value just meant to get any inherited value, too.

But that's what `t` is for right? Or I'm misunderstanding the documentation :thinking: 

If using `'default` it would then replace existing background (from my org block face) with that of `'default` so that wouldn't do it either. I've changed my Org block background to blue to better show what I mean:

Org buffer:
![image](https://user-images.githubusercontent.com/5608624/107768229-6d0a8680-6d36-11eb-93ba-0cf0f7c00bef.png)

Selectrum buffer:
![image](https://user-images.githubusercontent.com/5608624/107768610-02a61600-6d37-11eb-86f2-9f8b91c0aa6d.png)

Reply:
> If using 'default it would then replace existing background (from my org block face) with that of 'default so that wouldn't do it either.

This should only happen if your the `selectrum-current-candidate` face or the faces it inherits from doesn't specify a background. I tested it again with your theme and face config you posted and it seems this has to do with loading issues, when the `hl-line` face is not yet defined you can't inherit from it, you can test this by triggering the load with an activate of `hl-line-mode` afterwards it correctly gets the background.

Reply:
But I wonder why in you picture the empty line extends with a highlighted background.

Reply:
Note that I did not use your `custom-theme-set-face..` I just tried and it seems it doesn't seem to have an effect on the face definition. I simply used `(set-face-attribute 'selectrum-current-candidate nil :inherit 'hl-line)`. As I said I'm not very familiar with theming so I'm unsure what this is about or how your config might interfere with things. With the default config all themes I tested seem to behave well and clearly indicate the selection for:

```elisp

(completing-read "Check: " (list (propertize "this" 'face '(:background "red"))
                                 (propertize "out" 'face '(:background "green"))))
```

When the themes don't specify a background for the highlight face the default background will be used which is on purpose as this way you can still clearly tell which one is the selected when candidates use a background.

Reply:
![Screenshot from 2021-02-12 15-20-27](https://user-images.githubusercontent.com/13802849/107779650-f83f4880-6d45-11eb-8131-dd73f98d94be.png)

Reply:
I decided to undo the current "fix" for #425 for now as I see more down than upsides currently. Further discussion on how this case could be improved can happen there.

Reply:
I think my issues are coming from the way I'm customizing Zenburn in the first place:

```lisp
(let ((custom--inhibit-theme-enable nil))
  (zenburn-with-color-variables
    (custom-theme-set-faces 'zenburn ...)))
```

In addition to the fact that I'm resetting `'hl-line` background inside of that let expression too. I'll do more tests and post in the issue I created. Thanks again for the help!